client/src/App.js

```diff
@@ -1,19 +1,63 @@
-import React from "react";
-import { MuiThemeProvider } from "@material-ui/core";
-import { BrowserRouter, Route } from "react-router-dom";
+import React, { useCallback, useContext, useEffect } from 'react';
+import { MuiThemeProvider } from '@material-ui/core';
+import { BrowserRouter, Route } from 'react-router-dom';
+import { theme } from './themes/theme';
+import './App.css';
+import DashBoard from './pages/DashBoard';
+import api from './utils/api';
+import Navbar from './Components/Navbar';
+import Blog from './pages/Blog';
+import Faq from './pages/Faq';
+import Profile from './pages/Profile';
+import { StateProvider, store } from './context/store';
+import { AUTH_ERROR, USER_LOADED } from './context/types';
 
-import { theme } from "./themes/theme";
-import LandingPage from "./pages/Landing";
+function App() {
+  const { dispatch } = useContext(store);
+  const loadUser = useCallback(async () => {
+    try {
+      const res = await api.get('/users/auth');
 
-import "./App.css";
+      dispatch({
+        type: USER_LOADED,
+        payload: res.data,
+      });
+    } catch (err) {
+      dispatch({
+        type: AUTH_ERROR,
+      });
+    }
+  }, [dispatch]);
+
+  const setAuthToken = (token) => {
+    if (token) {
+      api.defaults.headers.common['Bearer'] = token;
+      localStorage.setItem('token', token);
+    } else {
+      delete api.defaults.headers.common['Bearer'];
+      localStorage.removeItem('token');
+    }
+  };
+
+  useEffect(() => {
+    if (localStorage.token) {
+      setAuthToken(localStorage.token);
```

Comment:
I noticed that this causes localStorage setItem to be called every time someone reopens the app - not a huge deal, just something to note and could be improved with some refactoring. Don't recommend to do it now, because you might need to refactor anyway once we have signup and login APIs working

Reply:
Yes sir, the conditions could be better.  I think I wrapped loadUser and line 44 in the if statement so it doesn't get used right now.